### PR TITLE
[docs] Clarify wording around adding new associatedtypes to protocols.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
cc Xi: I don't know if the API/ABI checkers already account for this, we might want to update them if they don't already do that.

### CODE REVIEW

File: docs/LibraryEvolution.rst [hunk 1]
```diff
@@ -578,7 +578,9 @@ There are very few safe changes to make to protocols and their members:
 - The ``@discardableResult`` and ``@warn_unqualified_access`` attributes may
   be added to or removed from a function requirement.
 - A new ``associatedtype`` requirement may be added (with the appropriate
-  availability), as long as it has a default implementation.
+  availability), as long as it has a default implementation. If the protocol
+  did not have one or more ``associatedtype`` requirements before the change,
+  then this is a `binary-compatible source-breaking change`.
```
Comment (harlanhaskins) [inline docs/LibraryEvolution.rst:583]: Should these be double-backticks?

Comment (varungandhi-apple) [reply docs/LibraryEvolution.rst:583]: I think this is right, double backticks are for formatting code, single backticks are for references. Elsewhere in this document:

https://github.com/apple/swift/blob/a0cdeb5fecf30cfb01d31b5b8a4360851bdac0b9/docs/LibraryEvolution.rst#L88-L89

https://github.com/apple/swift/blob/a0cdeb5fecf30cfb01d31b5b8a4360851bdac0b9/docs/LibraryEvolution.rst#L162-L163

**Review by nkcsgexi — APPROVED** [review]
It's great to document this!👍

**Review by harlanhaskins — APPROVED** [review]

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test and merge

**Pull request merged**